transport/grpc.go

```diff
@@ -128,11 +128,16 @@ func (t *grpcTransport) Call(ctx context.Context, request *Request) (*Response,
 
 	ctx, cancel := requestContextWithTimeout(ctx, request)
 	defer cancel()
+
+	var errs error
+
 	transportResponse, err := t.Outbound.Call(ctx, t.requestToYARPCRequest(request))
-	if err != nil {
-		return nil, err
-	}
-	return yarpcResponseToResponse(transportResponse)
+	errs = multierr.Append(errs, err)
+
+	r, err := yarpcResponseToResponse(transportResponse)
+	errs = multierr.Append(errs, err)
+
+	return r, errs
```

Comment:
why do we make this change? This implies `yarpcResponseToResponse` now has to be able to handle error cases of `transportResponse` now (nil, or some object with unknown state).

Reply:
Well, this because, previously, on error we should just bubble up error and ignore the response altogether.

Since I now want to have response headers to be properly build/populated, we always want to build a proper response object, and return it together with the error.

So, we want `yarpcResponseToResponse` always to be called, I think.